renderpyg/sprite.py

```diff
@@ -275,11 +279,12 @@ def set_frame(self, frame=0, duration=0, **kwargs):
 		self.frame = self.names.get(frame, int(frame))
 		try:
 			self.image = self.images[self.frame]
-		except:
+		except Exception as e:
+                        print(f"[Error] {type(e).__name__}: {e}")
 			self.image = self.images[0]
 		self.duration = duration
-		angle = kwargs.get('angle', None) 
-		self.angle = angle if angle != None else self.angle
```

Comment:
This won't work as intended

self.angle = kwargs.get(“angle”, self.angle) is probably a proper solution